src/graphql/queries/EntryValues.ts

```diff
@@ -1,22 +1,165 @@
+                            type: 'PATH',
+                            formatter: [],
+                            value: x.name + (x.referencedCollections.items.length === 1 ? getTitleProp(x.referencedCollections.items[0], otherCollections) : '.uri')
+                        }]
+                    };
+                }
+                return {
+                    type: 'KEYVALUE',
+                    value: x.name,
+                    formatter: [],
+                    subComponents: [value]
+                } as ComponentConfig;
+            })
+    );
+}
+
+function componentPathsToMap(paths: string[]): {} {
+    const result = {};
+    for (const path of paths) {
+        let cur = result;
+        let segments = path.split('.');
+        while (segments.length > 0) {
+            const segment = segments.shift()!;
+            if (!cur.hasOwnProperty(segment)) {
+                cur[segment] = segments.length > 0 ? {} : true;
+            }
+            cur = cur[segment];
+        }
+    }
+    return result;
+}
+
+function getPaths(components: ComponentConfig[], result: string[]): string[] {
+
+    for (const component of components) {
+        switch (component.type) {
+            case 'DIVIDER':
+                getPaths(component.subComponents || [], result);
+                break;
+            case 'TITLE':
+                getPaths(component.subComponents || [], result);
+                break;
+            case 'LITERAL':
+                break;
+            case 'PATH':
+                if (component.value !== undefined) {
+                    result.push(component.value);
+                }
+                break;
+            case 'IMAGE':
+                if (component.subComponents !== undefined) {
+                    getPaths(component.subComponents, result);
+                }
+                break;
+            case 'LINK':
+                if (component.subComponents !== undefined) {
+                    getPaths(component.subComponents, result);
+                }
+                break;
+            case 'KEYVALUE':
+                if (component.subComponents !== undefined) {
+                    getPaths(component.subComponents, result);
+                }
+                break;
+            default:
+                checkUnknownComponent(component);
+                return result;
+        }
+    }
+    return result;
+}
+
+function mapToQuery(map: {}, prefix: string): string {
+    const result: string[] = [];
+    for (const key in map) {
+        if (typeof map[key] === 'boolean') {
+            if (key === 'uri') {
+                result.push(key);
+            } else {
+                result.push(key + ' { value type }');
+            }
+        } else {
+            const subQuery = mapToQuery(map[key], prefix + '  ') + '\n';
+            result.push(key + ' {\n' + subQuery + prefix + '}');
+        }
+    }
+    return result.map(line => (prefix || '') + line).join('\n');
```

Comment:
There should always be a prefix available right? In that case you don't have to add this or statement for an empty string. If the prefix should be optional, I'd suggest to add a default value to the argument:

`function mapToQuery(map: {}, prefix: string = ''): string {`